### ISSUE TITLE
How to build the docs locally post recent changes?

### ISSUE THREAD

**Issue by Centrus007** [pymedphys/pymedphys#254]
Instructions [here ](https://pymedphys.com/developer/documentation.html)no longer work... :(

**Comment by SimonBiggs** [comment]
Yeah... :sweat:

**Comment by SimonBiggs** [comment]
The following should work:

```bash
yarn bootstrap:dev:all
sphinx-autobuild -W -p 7070 docs docs/_build/html
```

**Comment by SimonBiggs** [comment]
So, the instruction in the docs are still essentially up to date, it's just the packages need to be bootstrapped again.

**Comment by Centrus007** [comment]
This still doesn't work at work, but it's because I can't run `yarn`. I get the following output when i try:

```bash
yarn install v1.12.3
[1/4] Resolving packages...
[2/4] Fetching packages...
error Command failed.
Exit code: 128
Command: git
Arguments: ls-remote --tags --heads git://github.com/abe33/docco.git
Directory: C:\Users\mjenni03\PythonProjects\pymedphys
Output:
fatal: unable to connect to github.com:
github.com[0: 192.30.255.113]: errno=No error
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

I think the problem is that I can't access git://github.com/abe33/docco.git because SSH just doesn't work on my work PC (blocked by firewall)? So I clone everything via HTTPS...

**Comment by Centrus007** [comment]
Replacing `git://` with `https://` everywhere seems to work.

**Comment by Centrus007** [comment]
Scratch that, still lots of errors

```bash
(pmp) λ yarn bootstrap:dev:all
yarn run v1.12.3
$ yarn bootstrap:dev && pip install -e .[docs,testing,linting]
$ yarn && yarn nbstripout:install && yarn install:dev
[1/4] Resolving packages...
success Already up-to-date.
$ pip install nbstripout && nbstripout --install
Collecting nbstripout
  Using cached https://files.pythonhosted.org/packages/22/4c/c80719f5ca9c8b138d7d26f07f15e254a22cce47bce0e8afe6f809eb3448/nbstripout-0.3.5-py2.py3-none-any.whl
Collecting nbformat (from nbstripout)
  Using cached https://files.pythonhosted.org/packages/da/27/9a654d2b6cc1eaa517d1c5a4405166c7f6d72f04f6e7eea41855fe808a46/nbformat-4.4.0-py2.py3-none-any.whl
Collecting ipython-genutils (from nbformat->nbstripout)
  Using cached https://files.pythonhosted.org/packages/fa/bc/9bd3b5c2b4774d5f33b2d544f1460be9df7df2fe42f352135381c347c69a/ipython_genutils-0.2.0-py2.py3-none-any.whl
Collecting jsonschema!=2.5.0,>=2.4 (from nbformat->nbstripout)
  Using cached https://files.pythonhosted.org/packages/aa/69/df679dfbdd051568b53c38ec8152a3ab6bc533434fc7ed11ab034bf5e82f/jsonschema-3.0.1-py2.py3-none-any.whl
Collecting jupyter-core (from nbformat->nbstripout)
  Using cached https://files.pythonhosted.org/packages/1d/44/065d2d7bae7bebc06f1dd70d23c36da8c50c0f08b4236716743d706762a8/jupyter_core-4.4.0-py2.py3-none-any.whl
Collecting traitlets>=4.1 (from nbformat->nbstripout)
  Using cached https://files.pythonhosted.org/packages/93/d6/abcb22de61d78e2fc3959c964628a5771e47e7cc60d53e9342e21ed6cc9a/traitlets-4.3.2-py2.py3-none-any.whl
Requirement already satisfied: setuptools in c:\users\mjenni03\appdata\local\continuum\anaconda3\envs\pmp\lib\site-packages (from jsonschema!=2.5.0,>=2.4->nbformat->nbstripout) (41.0.1)
Collecting six>=1.11.0 (from jsonschema!=2.5.0,>=2.4->nbformat->nbstripout)
  Using cached https://files.pythonhosted.org/packages/73/fb/00a976f728d0d1fecfe898238ce23f502a721c0ac0ecfedb80e0d88c64e9/six-1.12.0-py2.py3-none-any.whl
Collecting pyrsistent>=0.14.0 (from jsonschema!=2.5.0,>=2.4->nbformat->nbstripout)
  Downloading https://files.pythonhosted.org/packages/2c/a7/8a50738eb27e204aa271abe170dec7bdbb07128ed892fb3a92f14a69bae3/pyrsistent-0.15.1.tar.gz (106kB)
     |████████████████████████████████| 112kB 2.2MB/s
Collecting attrs>=17.4.0 (from jsonschema!=2.5.0,>=2.4->nbformat->nbstripout)
  Using cached https://files.pythonhosted.org/packages/23/96/d828354fa2dbdf216eaa7b7de0db692f12c234f7ef888cc14980ef40d1d2/attrs-19.1.0-py2.py3-none-any.whl
Collecting decorator (from traitlets>=4.1->nbformat->nbstripout)
  Using cached https://files.pythonhosted.org/packages/5f/88/0075e461560a1e750a0dcbf77f1d9de775028c37a19a346a6c565a257399/decorator-4.4.0-py2.py3-none-any.whl
Building wheels for collected packages: pyrsistent
  Building wheel for pyrsistent (setup.py) ... done
  Stored in directory: C:\Users\mjenni03\AppData\Local\pip\Cache\wheels\93\03\43\ed0b3c9666e2001e2e6cce51efebdf3ce5e8edfa92966b1f6f
Successfully built pyrsistent
Installing collected packages: ipython-genutils, six, pyrsistent, attrs, jsonschema, decorator, traitlets, jupyter-core, nbformat, nbstripout
Successfully installed attrs-19.1.0 decorator-4.4.0 ipython-genutils-0.2.0 jsonschema-3.0.1 jupyter-core-4.4.0 nbformat-4.4.0 nbstripout-0.3.5 pyrsistent-0.15.1 six-1.12.0 traitlets-4.3.2
$ yarn lerna run pip:install:dev --concurrency 1 && pip install -e .
$ C:\Users\mjenni03\PythonProjects\pymedphys\node_modules\.bin\lerna run pip:install:dev --concurrency 1
lerna notice cli v3.13.3
lerna info Executing command in 13 packages: "yarn run pip:install:dev"
lerna ERR! yarn run pip:install:dev exited 1 in 'pymedphys_sphinxtheme'
lerna ERR! yarn run pip:install:dev stdout:
$ yarn bootstrap && pip install -e .
$ yarn --production && yarn webpack:prod
[1/4] Resolving packages...
[2/4] Fetching packages...
info fsevents@1.2.9: The platform "win32" is incompatible with this module.
info "fsevents@1.2.9" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
[4/4] Building fresh packages...
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

lerna ERR! yarn run pip:install:dev stderr:
error C:\Users\mjenni03\PythonProjects\pymedphys\packages\pymedphys_sphinxtheme\node_modules\bourbon-neat\node_modules\node-sass: Command failed.
Exit code: 1
Command: node scripts/build.js
Arguments:
Directory: C:\Users\mjenni03\PythonProjects\pymedphys\packages\pymedphys_sphinxtheme\node_modules\bourbon-neat\node_modules\node-sass
Output:
Building: C:\Program Files\nodejs\node.exe C:\Users\mjenni03\PythonProjects\pymedphys\packages\pymedphys_sphinxtheme\node_modules\node-gyp\bin\node-gyp.js rebuild --verbose --libsass_ext= --libsass_cflags= --libsass_ldflags= --libsass_library=
gyp info it worked if it ends with ok
gyp verb cli [ 'C:\\Program Files\\nodejs\\node.exe',
gyp verb cli   'C:\\Users\\mjenni03\\PythonProjects\\pymedphys\\packages\\pymedphys_sphinxtheme\\node_modules\\node-gyp\\bin\\node-gyp.js',
gyp verb cli   'rebuild',
gyp verb cli   '--verbose',
gyp verb cli   '--libsass_ext=',
gyp verb cli   '--libsass_cflags=',
gyp verb cli   '--libsass_ldflags=',
gyp verb cli   '--libsass_library=' ]
gyp info using node-gyp@3.8.0
gyp info using node@10.15.0 | win32 | x64
gyp verb command rebuild []
gyp verb command clean []
gyp verb clean removing "build" directory
gyp verb command configure []
gyp verb check python checking for Python executable "python2" in the PATH
gyp verb `which` failed Error: not found: python2
gyp verb `which` failed     at getNotFoundError (C:\Users\mjenni03\PythonProjects\pymedphys\packages\pymedphys_sphinxtheme\node_modules\which\which.js:13:12)
gyp verb `which` failed     at F (C:\Users\mjenni03\PythonProjects\pymedphys\packages\pymedphys_sphinxtheme\node_modules\which\which.js:68:19)
gyp verb `which` failed     at E (C:\Users\mjenni03\PythonProjects\pymedphys\packages\pymedphys_sphinxtheme\node_modules\which\which.js:80:29)
gyp verb `which` failed     at C:\Users\mjenni03\PythonProjects\pymedphys\packages\pymedphys_sphinxtheme\node_modules\which\which.js:89:16
gyp verb `which` failed     at C:\Users\mjenni03\PythonProjects\pymedphys\packages\pymedphys_sphinxtheme\node_modules\isexe\index.js:42:5
gyp verb `which` failed     at C:\Users\mjenni03\PythonProjects\pymedphys\packages\pymedphys_sphinxtheme\node_modules\isexe\windows.js:36:5
gyp verb `which` failed     at FSReqWrap.oncomplete (fs.js:154:21)
gyp verb `which` failed  python2 { Error: not found: python2
gyp verb `which` failed     at getNotFoundError (C:\Users\mjenni03\PythonProjects\pymedphys\packages\pymedphys_sphinxtheme\node_modules\which\which.js:13:12)
gyp verb `which` failed     at F (C:\Users\mjenni03\PythonProjects\pymedphys\packages\pymedphys_sphinxtheme\node_modules\which\which.js:68:19)
gyp verb `which` failed     at E (C:\Users\mjenni03\PythonProjects\pymedphys\packages\pymedphys_sphinxtheme\node_modules\which\which.js:80:29)
gyp verb `which` failed     at C:\Users\mjenni03\PythonProjects\pymedphys\packages\pymedphys_sphinxtheme\node_modules\which\which.js:89:16
gyp verb `which` failed     at C:\Users\mjenni03\PythonProjects\pymedphys\packages\pymedphys_sphinxtheme\node_modules\isexe\index.js:42:5
gyp verb `which` failed     at C:\Users\mjenni03\PythonProjects\pymedphys\packages\pymedphys_sphinxtheme\node_modules\isexe\windows.js:36:5
gyp verb `which` failed     at FSReqWrap.oncomplete (fs.js:154:21)
gyp verb `which` failed   stack:
gyp verb `which` failed    'Error: not found: python2\n    at getNotFoundError (C:\\Users\\mjenni03\\PythonProjects\\pymedphys\\packages\\pymedphys_sphinxtheme\\node_modules\\which\\which.js:13:12)\n    at F (C:\\Users\\mjenni03\\PythonProjects\\pymedphys\\packages\\pymedphys_sphinxtheme\\node_modules\\which\\which.js:68:19)\n    at E (C:\\Users\\mjenni03\\PythonProjects\\pymedphys\\packages\\pymedphys_sphinxtheme\\node_modules\\which\\which.js:80:29)\n    at C:\\Users\\mjenni03\\PythonProjects\\pymedphys\\packages\\pymedphys_sphinxtheme\\node_modules\\which\\which.js:89:16\n    at C:\\Users\\mjenni03\\PythonProjects\\pymedphys\\packages\\pymedphys_sphinxtheme\\node_modules\\isexe\\index.js:42:5\n    at C:\\Users\\mjenni03\\PythonProjects\\pymedphys\\packages\\pymedphys_sphinxtheme\\node_modules\\isexe\\windows.js:36:5\n    at FSReqWrap.oncomplete (fs.js:154:21)',
gyp verb `which` failed   code: 'ENOENT' }
gyp verb check python checking for Python executable "python" in the PATH
gyp verb `which` succeeded python C:\Users\mjenni03\AppData\Local\Continuum\anaconda3\envs\pmp\python.EXE
gyp ERR! configure error
gyp ERR! stack Error: Command failed: C:\Users\mjenni03\AppData\Local\Continuum\anaconda3\envs\pmp\python.EXE -c import sys; print "%s.%s.%s" % sys.version_info[:3];
gyp ERR! stack   File "<string>", line 1
gyp ERR! stack     import sys; print "%s.%s.%s" % sys.version_info[:3];
gyp ERR! stack                                ^
gyp ERR! stack SyntaxError: invalid syntax
gyp ERR! stack
gyp ERR! stack     at ChildProcess.exithandler (child_process.js:294:12)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at maybeClose (internal/child_process.js:962:16)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:251:5)
gyp ERR! System Windows_NT 6.1.7601
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\mjenni03\\PythonProjects\\pymedphys\\packages\\pymedphys_sphinxtheme\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild" "--verbose" "--libsass_ext=" "--libsass_cflags=" "--libsass_ldflags=" "--libsass_library="
gyp ERR! cwd C:\Users\mjenni03\PythonProjects\pymedphys\packages\pymedphys_sphinxtheme\node_modules\bourbon-neat\node_modules\node-sass
gyp ERR! node -v v10.15.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok
Build failed with error code: 1
error Command failed with exit code 1.
error Command failed with exit code 1.

lerna ERR! yarn run pip:install:dev exited 1 in 'pymedphys_sphinxtheme'
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by Centrus007** [comment]
Okay, scrap *all* of that. I didn't have `nodejs` installed... :sweat:

**Comment by Centrus007** [comment]
Looks like its running python 2 code... ??

**Comment by SimonBiggs** [comment]
:( so I've been on a Linux machine only for the past week and a bit, and all our testing within CI has been on Linux and Mac... seems like it doesn't work on Windows. Frustrating.

I'll fix it.

**Comment by SimonBiggs** [comment]
This here appears to be the issue:

https://github.com/JeremyEnglert/JointsWP/issues/317

**Comment by SimonBiggs** [comment]
The issue essentially is that the dependencies are very old for the readthedocs theme... so old, that they can't build on Windows... I'm trying to see how much I can increase the dependencies without breaking the site...

**Comment by SimonBiggs** [comment]
Hopefully https://github.com/pymedphys/pymedphys/pull/257 fixes this issue. Let me know. :).

... I also don't understand what you meant by ssh vs https...?

**Comment by Centrus007** [comment]
So I didn't have `nodejs` installed initially. When I didn't, the `yarn` command was trying to access a number of git repos via ssh. This doesn't work for me, since I have the following problem: https://stackoverflow.com/questions/15589682/ssh-connect-to-host-github-com-port-22-connection-timed-out

But installing `nodejs` removed that issue and instead I got the old dependcy issue above

**Comment by Centrus007** [comment]
Resolved by #257
